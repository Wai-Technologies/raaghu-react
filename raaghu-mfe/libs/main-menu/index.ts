export { default as MainMenu } from "./main-menu";
export {default as PageNotFound} from './pagenotfound';