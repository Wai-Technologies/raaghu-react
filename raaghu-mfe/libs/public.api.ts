export * from "./state-management/public.api";
export * from "./shared/public.api";