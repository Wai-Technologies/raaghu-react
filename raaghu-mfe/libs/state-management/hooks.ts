import { TypedUseSelectorHook , useDispatch , useSelector } from "react-redux";
import type { RootState , AppDispatch } from "./index";

export const useAppSelector : TypedUseSelectorHook<RootState> = useSelector;
export const useAppDispatch = () => useDispatch<AppDispatch>();