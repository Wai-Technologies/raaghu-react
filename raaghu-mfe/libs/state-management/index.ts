import tagAdminReducer from "./tag-admin/tag-admin-slice";
import fileDescriptorsReducer from "./file-descriptors/file-descriptors-slice";
import directoryDescriptorsReducer from "./directory-descriptors/directory-descriptors-slice";
import auditLogsReducer from "./audit-logs/audit-logs-slice";
import languageTextsReducer from "./language-texts/language-texts-slice";
import pollAdminReducer from "./poll-admin/poll-admin-slice";
import planAdminReducer from "./plan-admin/plan-admin-slice";
import roleReducer from "./role/role-slice";
import organizationUnitReducer from "./organization-unit/organization-unit-slice";
import permissionsReducer from "./permissions/permissions-slice";
import scopesReducer from "./scopes/scopes-slice";
import featuresReducer from "./features/features-slice";
import { useDispatch } from "react-redux";
import { persistReducer, persistStore } from "redux-persist";
import { configureStore, combineReducers } from "@reduxjs/toolkit";
import storage from "redux-persist/lib/storage";
import languagesReducer from "./languages/languages-slice";
import PageNotFoundReducer from "./pagenotfound/pagenotfound-slice";
import formsResponseReducer from "./formsresponse/formsresponse-slice";
import HomeReducer from "./home/home-slice";
import logoutReducer from "./logout/logout-slice";
import registerReducer from "./register/register-slice";
import blogPostReducer from "./blog-post/blog-post-slice";
import pagesReducer from "./pages/pages-slice";
import { forgotPasswordReducer } from "./forgot-password/forgotpassword-slice";
import editionReducer from "./edition/edition-slice";
import { subscriptionReducer } from "./subscription/subscription-slice";
import languageTextReducer from "./language-text/language-text-slice";
import organizationReducer from "./organization-tree/organization-tree-slice";
import rolesReducer from "./role/role-slice";
import ClaimTypeReducer from "./claim-type/claim-type-slice";
import securityLogsReducer from "./security-logs/security-logs-slice";
import userReducer from "./user/user-slice";
import settingsReducer from "./settings/settings-slice";
import FileManagementReducer from "./file-management/file-management-slice";
import hostReducer from "./host/host-slice";
import applicationsReducer from "./applications/applications-slice";
import urlForwardingReducer from "./url-forwarding/url-forwarding-slice";
import textTemplateReducer from "./text-template/text-template-slice";
import tenantReducer from "./tenant/tenant-slice";
import chatsReducer from "./chats/chats-slice";
import bloggerReducer from "./blogger/blogger-slice";
import formsReducer from "./forms/forms-slice";
import paymentPlansReducer from "./payment-plans/paymentPlans-slice";
import personalDataReducer from "./personal-data/personal-data-slice";
import commentsReducer from "./comments/comments-slice";
import tagsReducer from "./tags/tags-slice";
import myAccountReducer from "./my-account/my-account-slice";
import paymentRequestsReducer from "./payment-requests/paymentRequests-slice";
import newslettersReducer from "./newsletters/newsletters-slice";
import pollsReducer from "./polls/polls-slice";
import globalResourcesReducer from "./global-resources/globalResources-slice";
import menusReducer from "./menus/menus-slice";
import blogsReducer from "./Blogs/blogs-slice";
import configureReducer from "./config/config-slice";

const persistConfig = {
    key: "root",
    storage,
    blacklist: ["forgotPassword"],
};
const rootReducer = combineReducers({
    tagAdmin: tagAdminReducer,
    fileDescriptors: fileDescriptorsReducer,
    directoryDescriptors: directoryDescriptorsReducer,
    auditLogs: auditLogsReducer,
    languageTexts: languageTextsReducer,
    pollAdmin: pollAdminReducer,
    planAdmin: planAdminReducer,
    role: roleReducer,
    organizationUnit: organizationUnitReducer,
    permissions: permissionsReducer,
    features: featuresReducer,
    languages: languagesReducer,
    configureStore: configureReducer,
    PageNotFound: PageNotFoundReducer,
    formsResponse: formsResponseReducer,
    Home: HomeReducer,
    Logout: logoutReducer,
    register: registerReducer,
    blogPost: blogPostReducer,
    pages: pagesReducer,
    forgotPassword: forgotPasswordReducer,
    subscription: subscriptionReducer,
    edition: editionReducer,
    languagesText: languageTextReducer,
    organization: organizationReducer,
    roles: rolesReducer,
    claimTypes: ClaimTypeReducer,
    securityLogs: securityLogsReducer,
    applications: applicationsReducer,
    scopes: scopesReducer,
    textTemplate: textTemplateReducer,
    auditLog: auditLogsReducer,
    user: userReducer,
    settings: settingsReducer,
    tenant: tenantReducer,
    chats: chatsReducer,
    blogger: bloggerReducer,
    forms: formsReducer,
    fileManagement: FileManagementReducer,
    paymentPlans: paymentPlansReducer,
    urlForwarding: urlForwardingReducer,
    personalData: personalDataReducer,
    comments: commentsReducer,
    tags: tagsReducer,
    host: hostReducer,
    myaccount: myAccountReducer,
    paymentRequests: paymentRequestsReducer,
    polls: pollsReducer,
    globalResources: globalResourcesReducer,
    newsletters: newslettersReducer,
    menus: menusReducer,
    blogs: blogsReducer
});
const persistedReducer = persistReducer(persistConfig, rootReducer);

export const store = configureStore({
    reducer: { persistedReducer },
});
export const persistor = persistStore(store);
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export const useAppDispatch: () => AppDispatch = useDispatch;
