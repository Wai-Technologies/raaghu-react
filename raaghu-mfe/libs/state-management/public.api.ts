export * from "./tag-admin/tag-admin-slice";
export * from "./file-descriptors/file-descriptors-slice";
export * from "./directory-descriptors/directory-descriptors-slice";
export * from "./audit-logs/audit-logs-slice";
export * from "./language-texts/language-texts-slice";
export * from "./poll-admin/poll-admin-slice";
export * from "./plan-admin/plan-admin-slice";
export * from "./role/role-slice";
export * from "./organization-unit/organization-unit-slice";
export * from "./permissions/permissions-slice";
export * from "./applications/applications-slice";
export * from "./scopes/scopes-slice";
export * from "./features/features-slice";
export * from "./edition/edition-slice";
export * from "./languages/languages-slice";
export * from "./pagenotfound/pagenotfound-slice";
export * from "./formsresponse/formsresponse-slice";
export * from "./home/home-slice";
export * from "./register/register-slice";
export * from "./blog-post/blog-post-slice";
export * from "./pages/pages-slice";
export * from "./index";
export * from "./forgot-password/forgotpassword-slice";
export * from "./subscription/subscription-slice";
export * from "./claim-type/claim-type-slice";
export * from "./security-logs/security-logs-slice";
export * from "./user/user-slice";
export * from "./language-text/language-text-slice";
export * from "./settings/settings-slice";
export * from "./tenant/tenant-slice";
export * from "./chats/chats-slice";
export * from "./file-management/file-management-slice";
export * from "./url-forwarding/url-forwarding-slice";
export * from "./host/host-slice";
export * from "./my-account/my-account-slice";
export * from "./forms/forms-slice";
export * from "./polls/polls-slice";
export * from "./newsletters/newsletters-slice";
export * from "./menus/menus-slice";
export * from "./Blogs/blogs-slice";
export * from "./logout/logout-slice";
export * from "./config/config-slice";
