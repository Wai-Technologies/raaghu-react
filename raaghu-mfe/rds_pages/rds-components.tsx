export * from "../../raaghu-components/src/index";
