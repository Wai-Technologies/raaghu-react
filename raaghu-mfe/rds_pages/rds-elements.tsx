export * from "../../raaghu-elements/src/index";
