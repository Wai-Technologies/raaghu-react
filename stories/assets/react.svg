<svg width="84" height="80" viewBox="0 0 84 80" fill="none" xmlns="http://www.w3.org/2000/svg">
<path d="M41.9998 46.3032C45.3591 46.3032 48.0824 43.54 48.0824 40.1315C48.0824 36.7229 45.3591 33.9597 41.9998 33.9597C38.6404 33.9597 35.9171 36.7229 35.9171 40.1315C35.9171 43.54 38.6404 46.3032 41.9998 46.3032Z" fill="#61DAFB"/>
<path d="M41.9998 46.3032C45.3591 46.3032 48.0824 43.54 48.0824 40.1315C48.0824 36.7229 45.3591 33.9597 41.9998 33.9597C38.6404 33.9597 35.9171 36.7229 35.9171 40.1315C35.9171 43.54 38.6404 46.3032 41.9998 46.3032Z" stroke="#61DAFB" stroke-width="4"/>
<path d="M42 55.4964C64.0914 55.4964 82 48.5584 82 40C82 31.4415 64.0914 24.5035 42 24.5035C19.9086 24.5035 2 31.4415 2 40C2 48.5584 19.9086 55.4964 42 55.4964Z" stroke="#61DAFB" stroke-width="4"/>
<path d="M28.7734 47.7482C39.8191 67.1601 54.6952 79.4276 62 75.1484C69.3048 70.8692 66.2723 51.6637 55.2266 32.2518C44.1809 12.8398 29.3048 0.572317 22 4.85154C14.6952 9.13076 17.7277 28.3363 28.7734 47.7482Z" stroke="#61DAFB" stroke-width="4"/>
<path d="M28.7732 32.2518C17.7275 51.6637 14.695 70.8692 21.9998 75.1485C29.3046 79.4277 44.1807 67.1602 55.2264 47.7482C66.2721 28.3363 69.3046 9.1308 61.9998 4.85158C54.695 0.572351 39.8189 12.8398 28.7732 32.2518Z" stroke="#61DAFB" stroke-width="4"/>
</svg>
