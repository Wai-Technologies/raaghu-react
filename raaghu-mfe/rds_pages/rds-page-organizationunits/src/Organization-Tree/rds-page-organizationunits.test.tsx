import React from "react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import { Provider, useDispatch } from "react-redux";
import { store } from "../../../../libs/state-management";
import configureMockStore from "redux-mock-store";
import OrganizationTree from "./Organization-Tree";
import { RdsAlert, RdsButton, RdsIcon, RdsInput, RdsNavtabs, RdsOffcanvas } from "../../../rds-elements";
import { RdsCompAlertPopup, RdsCompDatatable, RdsCompOrganizationTree } from "../../../rds-components";
// Mock lodash-es directly in the test file
jest.mock("lodash-es", () => ({
  filter: jest.fn(),
  forEach: jest.fn(),
  // Add any other lodash functions that you are using in your code
}));
global.location = {
  pathname: "../../../../libs/proxy/core",
} as any;

jest.mock("react-router-dom", () => ({
  useNavigate: () => jest.fn(),
}));
jest.mock("react-i18next", () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));
jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
  useDispatch: jest.fn(),
}));
jest.mock("../../../../../raaghu-components/src/index.ts", () => ({
  ...jest.requireActual("../../../../../raaghu-components/src/index.ts"),
}));
jest.mock("../../../../../raaghu-elements/src/index.ts", () => ({
  ...jest.requireActual("../../../../../raaghu-elements/src/index.ts"),
}));
jest.mock("../../../../libs/state-management/index.ts", () => ({
  store: {
    getState: jest.fn().mockReturnValue({
      persistedReducer: {
        configureStore: {
          configuration: {
            auth: {
              grantedPolicies: {},
            },
          },
        },
      },
    }),
    dispatch: jest.fn(),
    subscribe: jest.fn(),
  },
}));

// Mock the rds-elements and rds-components
jest.mock("../../../../../raaghu-elements/src", () => ({
  RdsButton: jest.fn(),
  RdsOffcanvas: jest.fn(),
  RdsAlert: jest.fn(),
  RdsSearch: jest.fn(),
  RdsModal: jest.fn(),
  RdsNavtabs: jest.fn(),
  RdsCheckbox: jest.fn(),
  RdsIcon: jest.fn(),
  RdsInput: jest.fn(),
}));

jest.mock("../../../../../raaghu-components/src", () => ({
  RdsCompApiScopeBasicResource: jest.fn(),
  RdsCompDatatable: jest.fn(),
  RdsCompAlertPopup: jest.fn(),
  RdsOffcanvas: jest.fn(),
  RdsCompProfilePicture: jest.fn(),
  RdsCompPersonalInfo: jest.fn(),
  RdsCompChangePassword: jest.fn(),
  RdsCompOrganizationTree: jest.fn(),
}));

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useDispatch: jest.fn(),
}));

const mockStore = configureMockStore();

// Mock localStorage in your test setup
const localStorageMock: Storage = {
  getItem: jest.fn(),
  setItem: jest.fn(),
  removeItem: jest.fn(),
  clear: jest.fn(),
  length: 0, // Add length property
  key: jest.fn(), // Add key method
};

global.localStorage = localStorageMock;
describe("Organization page", () => {
  it("renders organization page correctly", () => {
    const mockDispatch = jest.fn();
    (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
    render(
      <Provider store={store}>
        <OrganizationTree />
      </Provider>
    );
    document.querySelector('[name="AbpIdentity.OrganizationTree"]');
    document.querySelector('[button="NEW ROOT UNIT"]');
    document.querySelector('[button="AbpIdentity.NewRole"]');
    document.querySelector('[canvasTitle="Select Member"]');
  });

it("Check organization page data", () => {
  const mockDispatch = jest.fn();
  const nodeColor = {
    color: "red",
    borderColor: "black",
  };
  const oData = {
    id: 1,
    parentId: 0,
    code: "code",
    displayName: "displayName",
    children: [],
  };
  const handlerDeleteNode = jest.fn();
  const handlerCreateSubUnit = jest.fn();
  const handlerSelectNode = jest.fn();
  const handlerNodeEdit = jest.fn();
  const handlerNewUser = jest.fn();
  const handlerNewRoles = jest.fn();
  const navtabsItems=
    {
      label: "Label 1",
      id: "1",
      value: "Value 1",
    };
    const handlerMemberActions = jest.fn();
    const handlerRoleActions = jest.fn();
    const tableHeadersMembers = { displayName: "Label 1", key: "1", datatype: "Value 1" };
    const tableHeadersRoles = { displayName: "Label 2", key: "2", datatype: "Value 2" };
    const handlerUserSelect = jest.fn();
    const handlerCloseOffcanvas = jest.fn();
    const handlerUserPush = jest.fn();
    const handlerRoleSelect = jest.fn();
    const handlerRolePush = jest.fn();
    const handlerAddNestedNode = jest.fn();
    const handlerEdit = jest.fn();
    const setVal = jest.fn();
    const handlerAddSubUnitNode = jest.fn();
    const handlerAddRootNode = jest.fn();
    const handlerDeleteConfirm = jest.fn();
    const handlerDeleteConfirmUserRoles = jest.fn();
    const inputReset =true;
  (useDispatch as jest.Mock).mockReturnValue(mockDispatch);
 const org = render(
    <Provider store={store}>
      <OrganizationTree />
      <RdsAlert alertmessage={""} colorVariant={""} />
      <RdsCompOrganizationTree nodeColor={[nodeColor]} organizationTreeData={[oData]} mutable={false}  onDeleteNode={handlerDeleteNode} onCreateSubUnit={handlerCreateSubUnit} onSelectNode={handlerSelectNode} onNodeEdit={handlerNodeEdit}  buttonLabel="NEW ROOT UNIT"/>
      <RdsIcon />
      <RdsButton label="New Member"  onClick={handlerNewUser} data-bs-dismiss="offcanvas"/>
      <RdsButton label="AbpIdentity.NewRole"  onClick={handlerNewRoles} />
      <RdsNavtabs navtabsItems={[navtabsItems]} type={"tabs"}/>
      <RdsCompDatatable tableHeaders={[tableHeadersMembers]} tableData={[]} pagination={false} onActionSelection={handlerMemberActions}/>
      <RdsCompDatatable tableHeaders={[tableHeadersRoles]} tableData={[]} pagination={false}  onActionSelection={handlerRoleActions}/>
      <RdsOffcanvas placement={"end"} backDrop={false} scrolling={false} offId={"oMember-add-off"} canvasTitle={"Select Member"} />
      <RdsCompDatatable tableHeaders={[tableHeadersMembers]} tableData={[]} pagination={true} onRowSelect={handlerUserSelect}/>
      <RdsButton label="AbpUi.Cancel" onClick={handlerCloseOffcanvas} databsdismiss="offcanvas"/>
      <RdsButton label="AbpUi.Save" onClick={handlerUserPush} databsdismiss="offcanvas"/>
      <RdsOffcanvas placement={"end"} backDrop={false} scrolling={false} offId={"oRole-add-off"} canvasTitle={"AbpIdentity.SelectRoles"} />
      <RdsCompDatatable tableHeaders={[tableHeadersRoles]} tableData={[]} pagination={true}  recordsPerPage={10}  onRowSelect={handlerRoleSelect}/>
     <RdsButton label="AbpUi.Cancel" databsdismiss="offcanvas"/>
     <RdsButton label="AbpUi.Save" onClick={handlerRolePush}/>
     <RdsOffcanvas placement={"end"} backDrop={true} scrolling={false} offId={"aoganization"} canvasTitle={"AbpIdentity.NewOrganizationUnit"}  onClose={handlerCloseOffcanvas}/>
    <RdsInput label="AbpIdentity.Name" placeholder="Enter Name"  onChange={(e) => setVal(e.target.value)}/>
    <RdsButton label="AbpUi.Save"  onClick={handlerAddNestedNode}/>
    <RdsOffcanvas placement={"end"} backDrop={true} scrolling={false} offId={"boganization"} canvasTitle={"Edit Organization Unit"} />
    <RdsInput label="AbpIdentity.Name" placeholder="Enter Name" onChange={(e) => setVal(e.target.value)}/>
    <RdsButton label="AbpUi.Save"  onClick={handlerEdit}/>
    <RdsOffcanvas placement={"end"} backDrop={false} scrolling={false} offId={"coganization"} canvasTitle={"Add Organization Sub Unit"}  onClose={handlerCloseOffcanvas} />
    <RdsInput label="Organization Name" placeholder="Enter Organization Name" onChange={(e) => setVal(e.target.value)} reset={inputReset}/>
    <RdsButton  onClick={handlerAddSubUnitNode}/>
    <RdsOffcanvas placement={"end"} backDrop={false} scrolling={false} offId={"doganization"} canvasTitle={"Add Organization New Unit"} onClose={handlerCloseOffcanvas} />
    <RdsInput label="Organization Name" placeholder="Enter Organization Name" onChange={(e) => setVal(e.target.value)}/>
    <RdsButton label="AbpUi.Save"  onClick={handlerAddRootNode}/>
    <RdsCompAlertPopup alertID={"deleteTreeNode"}  onSuccess={handlerDeleteConfirm}/>
    <RdsCompAlertPopup alertID={"orgMemRole"} onSuccess={handlerDeleteConfirmUserRoles}/>
    </Provider>
  );
  expect(org).toMatchSnapshot();
});

  afterEach(() => {
    jest.clearAllMocks(); // Clear mock calls between tests
    // Reset localStorage to a new object
    global.localStorage = {
      getItem: jest.fn(),
      setItem: jest.fn(),
      removeItem: jest.fn(),
      clear: jest.fn(),
      length: 0,
      key: jest.fn(),
    };
  });
});
