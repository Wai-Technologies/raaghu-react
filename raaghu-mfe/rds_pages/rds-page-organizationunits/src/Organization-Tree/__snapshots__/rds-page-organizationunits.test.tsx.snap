// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Organization page Check organization page data 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="container-fluid p-0 m-0"
        >
          <div
            class="row h-100"
          >
            <div
              class="col-xxl-6 col-xl-6 col-12 mb-xxl-0 mb-xl-0 mb-lg-0 mb-4 ps-lg-1 ps-xl-1 ps-xxl-1 col-lg-12 col-md-12 ps-lg-3 "
            >
              <div
                class="card rounded-0 card-full-stretch border-0"
              >
                <div
                  class="card-header bg-transparent"
                >
                  <h5
                    class="card-title fw-medium"
                  >
                    AbpIdentity.SelectAnOrganizationUnitToSeeMembers
                  </h5>
                </div>
                <div />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="container-fluid p-0 m-0"
      >
        <div
          class="row h-100"
        >
          <div
            class="col-xxl-6 col-xl-6 col-12 mb-xxl-0 mb-xl-0 mb-lg-0 mb-4 ps-lg-1 ps-xl-1 ps-xxl-1 col-lg-12 col-md-12 ps-lg-3 "
          >
            <div
              class="card rounded-0 card-full-stretch border-0"
            >
              <div
                class="card-header bg-transparent"
              >
                <h5
                  class="card-title fw-medium"
                >
                  AbpIdentity.SelectAnOrganizationUnitToSeeMembers
                </h5>
              </div>
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
